### ISSUE TITLE
Feature Request: Recursive Folder Embedding for Configuration Files

### ISSUE THREAD

**Issue by JakePeralta7** [DFIR-ORC/dfir-orc#73]
Description: I would like to request a feature that allows embedding an entire folder of configuration files for a specific tool without needing to mention each file individually. This feature should support recursive embedding, meaning it should include all files and subfolders within the specified folder.

Use Case: This feature would be particularly useful for managing large sets of configuration files, making it easier to maintain and update them without having to list each file separately. It would streamline the process and reduce the potential for errors.

**Comment by maxdal89** [comment]
did you manage to find a solution? I should integrate an entire folder but I can't.

**Comment by JakePeralta7** [comment]
> did you manage to find a solution? I should integrate an entire folder but I can't.

Hi!
I didn't manage to find an elegant solution for that (creating a script that would receive a folder and create the appropriate XML crossed my mind though)

Are you trying to embed KAPE?

**Comment by maxdal89** [comment]
no, i would like to include a sigma based triaging tool, like Hayabusa which needs the rules folder.

 any ideas?

 i was trying zircolite with a single sigma rule file

**Comment by JakePeralta7** [comment]
> no, i would like to include a sigma based triaging tool, like Hayabusa which needs the rules folder.
> 
> any ideas?
> 
> i was trying zircolite with a single sigma rule file

Are you planning to run them on the investigated machine? or in the backend as part of the post-processing of the artifacts

**Comment by maxdal89** [comment]
The idea is to do both:
-run a triaging tool (hayabusa,APT-Hunter or Zircolite) directly on the machine, so in case of many hosts I can distribute the workload
-acquire a basic set of data, like that of Kape Triage to be processed by a server in my forensic lab.

**Comment by JakePeralta7** [comment]
> The idea is to do both: -run a triaging tool (hayabusa,APT-Hunter or Zircolite) directly on the machine, so in case of many hosts I can distribute the workload -acquire a basic set of data, like that of Kape Triage to be processed by a server in my forensic lab.

If you are looking to scale and distribute, shouldn't you use Velociraptor or GRR?

And for scaling post-processing in the backend, you can deploy OpenRelik on k8s (they've containerized hayabusa for example to be a worker).

idk, feels kind of wrong to process EVTX files on the investigated endpoint.

**Comment by maxdal89** [comment]
Why wrong to process EVTX files on the investigated endpoint?
